yt_dlp/extractor/generic.py

```diff
@@ -2205,6 +2233,23 @@ def _extra_manifest_info(self, info, manifest_url):
             for fmt in self._downloader._get_formats(info):
                 fmt['url'] = update_url_query(fmt['url'], query)
 
+        # Attempt to detect live HLS or set VOD duration
+        m3u8_url = traverse_obj(self._downloader._get_formats(info), (
+            lambda _, v: v['protocol'] == 'm3u8_native', 'url'), get_all=False)
+        if m3u8_url:
+            is_live = self._configuration_arg('is_live', [None])[0]
+            if is_live is not None:
+                info['live_status'] = 'not_live' if is_live == 'false' else 'is_live'
+                return
+            headers = traverse_obj(info, (
+                (None, ('formats', ...)), 'http_headers', {dict}), get_all=False) or {}
+            duration = self._extract_m3u8_vod_duration(
+                m3u8_url, info.get('id'), headers=headers, note='Checking m3u8 live status',
```

Comment:
Different formats can have different headers.
```suggestion
        m3u8_format = next(f for f in self._downloader._get_formats(info) if determine_protocol(f) == 'm3u8_native')
        if m3u8_format:
            is_live = self._configuration_arg('is_live', [None])[0]
            if is_live is not None:
                info['live_status'] = 'not_live' if is_live == 'false' else 'is_live'
                return
            headers = m3u8_format.get('http_headers') or info.get('http_headers')
            duration = self._extract_m3u8_vod_duration(
                m3u8_format['url'], info['id'], headers=headers, note='Checking m3u8 live status',
```